annotation/tests/test_tasks_services.py

```diff
@@ -0,0 +1,343 @@
+    with patch("sqlalchemy.orm.Session", spec=True) as mock_session:
+        db_session = mock_session()
+        task_info = {"user_id": 1, "file_id": 2, "job_id": 3, "pages": {4, 5}}
+        existing_pages = [4, 5]
+
+        mock_query = db_session.query.return_value
+        mock_query.filter.return_value.all.return_value = [(existing_pages,)]
+
+        with pytest.raises(
+            FieldConstraintError, match=".*tasks for this user: {4, 5}.*"
+        ):
+            check_cross_annotating_pages(db_session, task_info)
+
+
+@pytest.mark.parametrize(
+    (
+        "failed",
+        "annotated",
+        "annotation_user",
+        "validation_user",
+        "expected_error_message_pattern",
+    ),
+    (
+        (
+            {1},
+            set(),
+            False,
+            True,
+            r"Missing `annotation_user_for_failed_pages` param",
+        ),
+        (
+            set(),
+            {2},
+            True,
+            False,
+            r"Missing `validation_user_for_reannotated_pages` param",
+        ),
+    ),
+)
+def test_validate_user_actions_missing_users(
+    failed: set,
+    annotated: set,
+    annotation_user: bool,
+    validation_user: bool,
+    expected_error_message_pattern: str,
+):
+    with pytest.raises(HTTPException) as excinfo:
+        validate_user_actions(
+            is_validation=True,
+            failed=failed,
+            annotated=annotated,
+            not_processed=set(),
+            annotation_user=annotation_user,
+            validation_user=validation_user,
+        )
+    assert re.match(expected_error_message_pattern, excinfo.value.detail)
+
+
+@pytest.mark.parametrize(
+    (
+        "failed",
+        "annotated",
+        "not_processed",
+        "expected_error_message_pattern",
+    ),
+    (
+        (
+            set(),
+            set(),
+            set(),
+            r"Validator did not mark any pages as failed",
+        ),
+        (
+            {1},
+            set(),
+            set(),
+            r"Validator did not edit any pages",
+        ),
+        (
+            {1},
+            {2},
+            {3},
+            r"Cannot finish validation task. There are not processed pages",
+        ),
+    ),
+)
+def test_validate_user_actions_invalid_states(
+    failed: set,
+    annotated: set,
+    not_processed: set,
+    expected_error_message_pattern: str,
+):
+    with pytest.raises(HTTPException) as excinfo:
+        validate_user_actions(
+            is_validation=True,
+            failed=failed,
+            annotated=annotated,
+            not_processed=not_processed,
+            annotation_user=True,
+            validation_user=True,
+        )
+    assert re.match(expected_error_message_pattern, excinfo.value.detail)
+
+
+def test_create_annotation_task_success():
+
+    with patch("sqlalchemy.orm.Session", spec=True) as mock_session, patch(
+        "annotation.jobs.services.update_user_overall_load"
```

Comment:
```suggestion
        "annotation.tasks.services.update_user_overall_load"
```
Wrong patch, this error creates bad assertions.